pkg/controller/controller_test.go

```diff
@@ -104,7 +105,10 @@ func TestHandlePopup(t *testing.T) {
 			t.Parallel()
 
 			harness := envtest.NewServerConfig(t, "9091", tc.allowlist, true)
-			c := New(harness.Renderer, harness.Processor, tc.allowlist)
+			c, err := New(context.Background(), harness.Renderer, harness.Processor, tc.allowlist)
```

Comment:
ctx is defined on L60/61. You can use that here.

Reply:
updated in the latest version.